test/suite/test_compact11.py

```diff
@@ -0,0 +1,190 @@
+            time.sleep(0.1)
+
+    def compare_bitmap(self, orig, new):
+        # Compare the bitmaps from the metadata. Once a bit is set, it should never
+        # be cleared. But new bits could be set. So the check is only: if the original
+        # bitmap has a bit set then the current bitmap must be set for that bit also.
+        #
+        # First convert both bitmaps to a binary string, accounting for any possible leading
+        # zeroes (that would be truncated off). Then compare bit by bit.
+        orig_bits = bin(int('1'+orig, 16))[3:]
+        new_bits = bin(int('1'+new, 16))[3:]
+        self.pr("Original bitmap in binary: " + orig_bits)
+        self.pr("Reopened bitmap in binary: " + new_bits)
+        for o_bit, n_bit in zip(orig_bits, new_bits):
+            if o_bit != '0':
+               self.assertTrue(n_bit != '0')
+
+    def parse_blkmods(self, uri):
+        meta_cursor = self.session.open_cursor('metadata:')
+        config = meta_cursor[uri]
+        meta_cursor.close()
+        # The search string looks like: ,blocks=feffff1f000000000000000000000000
+        # Obtain just the hex string.
+        b = re.search(',blocks=(\w+)', config)
+        self.assertTrue(b)
+        # The bitmap string after the = is in group 1.
+        blocks = b.group(1)
+        self.pr(f"parse_blkmods for '{uri}': {blocks}")
+        return blocks
+
+    def test_compact11(self):
+        # Create N tables.
+        uris = []
+        files = []
+        bitmaps = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            files.append(f'file:test_compact11_{i}.wt')
+            self.session.create(uri, self.create_params)
+
+        # Populate the first half of each table.
+        for uri in uris:
+            self.populate(uri, 0, self.table_numkv // 2, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Take a full backup that will be used for incremental backups later during the test.
+        os.mkdir(self.backup_incr)
+        self.initial_backup = True
+        self.take_full_backup(self.backup_incr)
+        self.initial_backup = False
+
+        # Insert the latter 50% in each table.
+        for uri in uris:
+            self.populate(uri, self.table_numkv // 2, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of each file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Generate the bitmaps of each file.
+        for uri in files:
+            bitmaps.append(self.parse_blkmods(uri))
+
+        # Turn on background compaction to allow the bitmap blocks to be modified
+        # from compact operation.
+        self.turn_on_bg_compact('free_space_target=1MB')
+        while self.get_files_compacted(uris) < self.num_tables:
```

Comment:
I don't know how many times we loop here, but maybe take an incremental backup here so that we get multiple backups loops through here and can get multiple iterations.

Reply:
Unfortunately we can't take multiple incremental backups when we are test for the bitmap. The reason is because each incremental backup's bitmap will only show modified relative to the previous bitmap. What that means is that the compare_bitmaps will between two incremental bitmaps will not work, and there is no way of testing it. Let me provide an example to explain this better. E.g.
```
Base backup bitmap = 0000
1st Incr backup bitmap = 0011
2nd incr backup bitmap = 1100
```
If we see here, 2nd incremental bitmap shows that the first two bits are modified **compared to** the first incremental backup. There is nothing that we can test between the 2nd incremental and 1st incremental.

Reply:
It could work if each incremental used the original/first backup ID as the source instead of moving the source forward. For example, if the first full backup enabling incremental used "ID1", the first incremental could use "src=ID1,this=ID2" then the next "src=ID1,this=ID3", ...

I don't know if the python functions are set up this way, but it should work from WiredTiger's perspective and fix the bitmap comparison issue.